Bundle viewer

bnmajor opened the pull request:
Bundle the itk-vtk-viewer with the python package

WIP: Relies on Kitware/itk-vtk-viewer#475. Will need to update viewer version once that PR is in.

Fixes #53

bnmajor committed:
build(bundle itk-vtk-viewer): Bundle itk-vtk-viewer in package

bnmajor committed:
chore(pathlib): Use pathlib instead of os.path

thewtex commented on client/src/ImageSelector.jsx:
```diff
@@ -8,6 +8,7 @@ import {
 } from './utils'
 import './style.css'
 import { debounce } from 'lodash'
+import 'itk-vtk-viewer'
```
-> something like: 

```
import createViewer from 'itk-vtk-viewer/src/createViewer'
```

The idea: we do not use the exports of the package (which is pointing to a bundled version). Avoid the current difficulties bundling the bundle. In the future we will have nice ESM to work with. For now, we will also have to set up the WebPack rules similar to itk-vtk-viewer: 1) a `resolve` `alias` for `itkConfig.js`. 2) The vtk.js WebPack rules for building the shaders, CSS, etc.

CC @PaulHax

bnmajor commented:
Thanks for the tips @thewtex! I have been working on this some more using your suggestions but I am currently stuck on the following error:
```
ERROR in ./node_modules/itk-vtk-viewer/src/ViewerStore.js
Module build failed (from ./node_modules/babel-loader/lib/index.js):
SyntaxError: /home/local/KHQ/brianna.major/tensorboard-plugin-3d/client/node_modules/itk-vtk-viewer/src/ViewerStore.js: Support for the experimental syntax 'decorators-legacy' isn't currently enabled (23:3):

  21 | class MainUIStore {
  22 |   uiContainer = null
> 23 |   @observable collapsed = false
     |   ^
  24 |   @observable annotationsEnabled = true
  25 |   @observable axesEnabled = false
  26 |   @observable fullscreenEnabled = false

```
I've pushed a WIP commit with the most recent changes in case there is something obvious that you can see that I am doing wrong, in the meantime I'll keep looking into what's going on and reading up on webpack rules to try and understand what I'm missing.

thewtex commented:
> `@observable`

@bnmajor I believe this is addressed by enabling the `mobx` babel preset:

https://github.com/Kitware/itk-vtk-viewer/blob/e0b5bffc8cd499b9f8ac310ba2b1fe7ee3309a9b/.babelrc#L9

bnmajor commented:
> > `@observable`
> 
> @bnmajor I believe this is addressed by enabling the `mobx` babel preset:
> 
> https://github.com/Kitware/itk-vtk-viewer/blob/e0b5bffc8cd499b9f8ac310ba2b1fe7ee3309a9b/.babelrc#L9

Thanks @thewtex! I did try this but still have the error, unfortunately.

thewtex commented:
@bnmajor is `babel-preset-mobx` installed?

https://github.com/Kitware/itk-vtk-viewer/blob/e0b5bffc8cd499b9f8ac310ba2b1fe7ee3309a9b/package.json#L66


Or is a different babel config being picked up?

bnmajor committed:
build(babel): Rename .babelrc -> babel.config.json

Use babel.config rather than .babelrc so that configuration is applied project
wide rather than to a subset of the files.

bnmajor commented:
> @bnmajor is `babel-preset-mobx` installed?
> 
> https://github.com/Kitware/itk-vtk-viewer/blob/e0b5bffc8cd499b9f8ac310ba2b1fe7ee3309a9b/package.json#L66
> 
> Or is a different babel config being picked up?

@thewtex Thank you for pointing this out, I had `babel-preset-mobx` installed but it was missing from the [webpack list of presets](https://github.com/KitwareMedical/tensorboard-plugin-3d/blob/50c0ee1bc5d0a97706f893f59d6764ac28315c2b/client/webpack.config.js#L28). The last change that I was missing was to rename the babel config from `.babelrc` to `babel.config.json` in order to apply the configuration to the entire project. Thank you for all of your help, I believe this is ready for review now!

thewtex reviewed (approved):
:sparkles: :sparkling_heart:
